omniisaacgymenvs/tasks/MFP2D_Virtual_IMU.py

```diff
@@ -0,0 +1,618 @@
+            thrust_cmds = self.actions.float()
+        elif self._discrete_actions == "Continuous":
+            # Transform continuous actions to [0, 1] discrete actions.
+            thrust_cmds = torch.clamp((self.actions + 1) / 2, min=0.0, max=1.0)
+        else:
+            raise NotImplementedError("")
+
+        # Applies the thrust multiplier
+        thrusts = self.virtual_platform.thruster_cfg.thrust_force * thrust_cmds
+        # Adds random noise on the actions
+        thrusts = self.AN.add_noise_on_act(thrusts)
+        # clear actions for reset envs
+        thrusts[reset_env_ids] = 0
+        # If split thrust, equally shares the maximum amount of thrust across thrusters.
+        if self.split_thrust:
+            factor = torch.max(
+                torch.sum(self.actions, -1),
+                torch.ones((self._num_envs), dtype=torch.float32, device=self._device),
+            )
+            self.positions, self.forces = self.virtual_platform.project_forces(
+                thrusts / factor.view(self._num_envs, 1)
+            )
+        else:
+            self.positions, self.forces = self.virtual_platform.project_forces(thrusts)
+        # Apply forces
+        self.apply_forces()
+        return
+
+    def apply_forces(self) -> None:
+        """
+        Applies all the forces to the platform and its thrusters."""
+
+        self._platforms.thrusters.apply_forces_and_torques_at_pos(
+            forces=self.forces, positions=self.positions, is_global=False
+        )
+        floor_forces = self.UF.get_floor_forces(self.root_pos)
+        torque_disturbance = self.TD.get_torque_disturbance(self.root_pos)
+        self._platforms.base.apply_forces_and_torques_at_pos(
+            forces=floor_forces,
+            torques=torque_disturbance,
+            positions=self.root_pos,
+            is_global=True,
+        )
+
+    def post_reset(self):
+        """
+        This function implements the logic to be performed after a reset."""
+
+        # implement any logic required for simulation on-start here
+        self.root_pos, self.root_rot = self._platforms.get_world_poses()
+        self.root_velocities = self._platforms.get_velocities()
+        self.dof_pos = self._platforms.get_joint_positions()
+        self.dof_vel = self._platforms.get_joint_velocities()
+
+        self.initial_root_pos, self.initial_root_rot = (
+            self.root_pos.clone(),
+            self.root_rot.clone(),
+        )
+        self.initial_pin_pos = self._env_pos
+        self.initial_pin_rot = torch.zeros(
+            (self.num_envs, 4), dtype=torch.float32, device=self._device
+        )
+        self.initial_pin_rot[:, 0] = 1
+
+        # control parameters
+        self.thrusts = torch.zeros(
+            (self._num_envs, self._max_actions, 3),
+            dtype=torch.float32,
+            device=self._device,
+        )
+
+        self.set_targets(self.all_indices)
+        self.imu.reset(self._num_envs)
```

Comment:
Post reset is global only, and in 2023.1.1 it's unclear when it's being called. (at least once at the beginning, but then?)